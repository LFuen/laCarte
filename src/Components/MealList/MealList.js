import React, {Component} from 'react'
import LaCarteContext from '../../context/LaCarteContext'
import {Link} from 'react-router-dom'
import './MealList.css'

class MealList extends Component {

    static contextType = LaCarteContext

    render() {
        const {meals} = this.context
        return (
            <div className='mealList'>
                {meals.map(meal => { return (
                    <Link to={`/meals/${meal.id}`} key={meal.id}>
                        <div className='polaroid'>
                            <span title='Pick Me!'>
                                <img className='chefMoji' alt={meal.meal_name} src={meal.img_url}/>
                            </span>
                            <div className='container'>
                                {<p>{meal.meal_name}</p>}
                            </div>
                        </div>
                    </Link>
                )})}
            </div>
        )
    }
}

export default MealList



