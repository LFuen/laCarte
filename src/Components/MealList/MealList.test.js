import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import {BrowserRouter} from 'react-router-dom'
import LaCarteContext from '../../context/LaCarteContext'
import MealList from './MealList'



describe(`MealList Component`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const list = renderer.create(<BrowserRouter><MealList/></BrowserRouter>)
                expect(list).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div');
            const renderer = new ShallowRenderer()

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            renderer.render(
                <BrowserRouter>
                    <LaCarteContext.Provider value={value}>
                        <MealList />
                    </LaCarteContext.Provider>                
                </BrowserRouter>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})