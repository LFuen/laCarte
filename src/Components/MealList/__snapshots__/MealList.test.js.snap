// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MealList Component Snapshot renders the UI as expected 1`] = `
<div
  className="mealList"
/>
`;
